### PR TITLE
deployment: Use statefulset for deploying the workload

### PR DESCRIPTION
**Pull request by avisiedo**
- A Statefulset object will be generated.
- The PVC information is included into the Statefulset
  object.
- For ephemeral storage, the PVC information will be
  empty and the /data volume will be an emptyDir
  into the Pod template embedded into the
  Statefulset object.
    
Signed-off-by: Alejandro Visiedo <avisiedo@redhat.com>

### CODE REVIEW

**Comment by vashirov** [comment]
I see that #34 and ths PR contain same commits. Is it possible to split them to make review easier or they depend on each other?

**Comment by avisiedo** [comment]
> I see that #34 and ths PR contain same commits. Is it possible to split them to make review easier or they depend on each other?

This depends on #34 . I have to rebase this after merging #34

File: README.md [hunk 1]
```diff
@@ -57,15 +57,28 @@ Experimental freeipa-operator for Freeipa.
 
 1. Now create a new namespace by: `kubectl create namespace my-freeipa`
 
-1. And create a new idm resource by: `make recreate-sample-idm`
+1. And create a new idm resource by:
+   `PASSWORD=myPassword124 SAMPLE=ephimeral-storage make sample-create`
```
Comment (vashirov) [inline README.md]: Typo s/ephimeral/ephemeral/g

File: HELP [hunk 2]
```diff
@@ -15,8 +15,9 @@ Subcommands:
 
   install-crds         Install the CRD manifest into the cluster.
   uninstall-crds       Uninstall the CRD manifest.
-  deploy-cluster       Deploy controller into the cluster.
+  deploy-cluster       Deploy controller into the cluster and CRDS.
```
Comment (vashirov) [inline HELP]: Nit: "Deploy controller and CRDs into the cluster"

File: manifests/statefulset_test.go [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+		Context("an IDM with PVC information", func() {
+			var idm = v1alpha1.IDM{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+					Labels: map[string]string{
+						"app": "freeipa",
+					},
+				},
+				Spec: v1alpha1.IDMSpec{
+					Realm:          "TEST.COM",
```
Comment (vashirov) [inline manifests/statefulset_test.go]: We should use IPA.TEST, as it has a special TLD for testing purposes. See https://www.rfc-editor.org/rfc/rfc2606.html#section-2 test.com is someone else's domain and we shouldn't test against it.

**Review by vashirov — COMMENTED** [review]
Besides minor issues, LGTM.

File: README.md (hunk 1 above)
Comment (avisiedo) [reply README.md]: Thanks! Updated!

File: HELP (hunk 2 above)
Comment (avisiedo) [reply HELP]: Thanks! Updated!

File: manifests/statefulset_test.go (hunk 3 above)
Comment (avisiedo) [reply manifests/statefulset_test.go]: Thanks for referencing it! updated!

**Comment by avisiedo** [comment]
Merging

**Pull request merged**